packages/commerce-sdk-react/src/hooks/ShopperBaskets/useShopperBasketAction.ts

```diff
@@ -0,0 +1,15 @@
+import {ActionResponse} from '../../types'
+import {ShopperBasketActions} from './types'
+
+// Q: how do we link the return types for actions?
+type execute = () => Promise<any>
```

Comment:
@kevinxh In a similar vain to Will's comment. We'll need generics to be able to handle scenarios where partners have SCAPI/OCAPI hooks set up to add custom properties to their responses. For a hook like  `useShopperBasketAction` its execute return value should be generic so it can be of type `CustomBasket` vs the default `Basket` type. 

I'm not 100% sure how that would be done, especially when the execute function is the function that we are concerned about having a custom type.. and it's only a property of the hooks return value. 

Could be tricky. But I'm not familiar enough with generics to know if thats hard or not.

Reply:
If we know `execute`'s return value to be of `Basket` type, but it's possible that what's returned from the API may have additional custom properties, then we can still use the Basket type (and not some CustomBasket). I just found out that Typescript supports [duck typing](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes.html#structural-type-system).

For example: try having this code and verify that there's no error.

```ts
interface Product {
    name: string
}
interface CustomProduct extends Product {
    size: number
}

const useProduct = (): Product => {
    return fetchProduct()
}

const fetchProduct = (): CustomProduct => {
    return {
        name: 'some product',
        size: 10
    }
}
```

Reply:
This is what it looks like in the editor:
<img width="1143" alt="Screen Shot 2022-07-13 at 12 21 00 PM" src="https://user-images.githubusercontent.com/10948652/178814561-fb83d0d4-f394-4c97-86be-f2f5c716d8f9.png">

Reply:
@bendvc for you question regarding custom properties `c_`. This is already built by `commerce-sdk-isomorphic` types.

From what I can see, the types from the library are all look like this:
```
type Basket = {
    adjustedMerchandizeTotalTax?: number;
    adjustedShippingTotalTax?: number;
    agentBasket?: boolean;
    basketId?: string;
    billingAddress?: OrderAddress;
    bonusDiscountLineItems?: Array<BonusDiscountLineItem>;
    channelType?: string;
    couponItems?: Array<CouponItem>;
    creationDate?: any;
    currency?: string;
...
} & {
    [key: string]: any;
};
```

notice the bottom part:
```
& {
    [key: string]: any;
}
```

This allows you to include any custom property in the types.

Reply:
Those overloads work, but they're kind of a nuisance because of the way enums work. If we use a plain object instead (`const actions = { ... } as const`), then we can simplify it to a simple generic. Not sure which is better, at this point.